extension/src/project/readOnlyDocument.ts

```diff
@@ -82,6 +82,22 @@ export class ReadonlyDocument implements vscode.TextDocument {
         return ret;
     }
 
+    // Added this to essentially cast standard TextDocument's to a ROD type enabling work to be done with enhanced standardized/functionality
+    // Related to to discussion issue#17
+    static getMemoryDocument(doc: vscode.TextDocument): ReadonlyDocument {        
+        let ret = new ReadonlyDocument('');
+        ret.eol = vscode.EndOfLine.CRLF;
+        ret.eolLength = 2;
+        ret.lineCount = doc.lineCount;
+        ret.languageId = doc.fileName.toUpperCase().slice(-4) === ".LSP" ? "autolisp" : "";
```

Comment:
Despite of the real purpose of this method, the post fix could be .PRJ,  .DCL, etc.
https://github.com/Autodesk-AutoCAD/AutoLispExt/blob/main/package.json#L281
https://github.com/Autodesk-AutoCAD/AutoLispExt/blob/main/package.json#L292
https://github.com/Autodesk-AutoCAD/AutoLispExt/blob/main/package.json#L301